File: src/main/java/net/fellter/moreLeafParticles/yacl/ModMenuImpl.java
package net.fellter.moreLeafParticles.yacl;

import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;
import com.terraformersmc.modmenu.api.UpdateChecker;
import net.fellter.moreLeafParticles.MoreLeafParticles;

public class ModMenuImpl implements ModMenuApi {
	@Override
	public ConfigScreenFactory<?> getModConfigScreenFactory() {
		return MoreLeafParticles.isYACLPresent()
				? parentScreen -> YACLImpl.create().generateScreen(parentScreen)
				: ModMenuApi.super.getModConfigScreenFactory();
	}

	@Override
	public UpdateChecker getUpdateChecker() {
		return new MoreLeafParticlesUpdateChecker();
	}
}

File: src/main/java/net/fellter/moreLeafParticles/config/ModMenuImpl.kt
package net.fellter.moreLeafParticles.config

import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import com.terraformersmc.modmenu.api.UpdateChecker
import net.fellter.moreLeafParticles.MoreLeafParticles.Companion.isYACLPresent
import net.minecraft.client.gui.screens.Screen

class ModMenuImpl : ModMenuApi {
	override fun getModConfigScreenFactory(): ConfigScreenFactory<*> {
		return if (isYACLPresent)
			ConfigScreenFactory { parentScreen: Screen -> YACLImpl.create().generateScreen(parentScreen) }
		else
			super.getModConfigScreenFactory()
	}

	override fun getUpdateChecker(): UpdateChecker = MoreLeafParticlesUpdateChecker()
}
